pymongoexplain/__main__.py

```diff
@@ -14,47 +14,170 @@
 
 
 from pymongo import monitoring, MongoClient
-from pymongo.collection import Collection
+from pymongo.collection import Collection, ReturnDocument
+from pymongo.cursor import CursorType
+from pymongo.common import BaseObject
+from .explainable_collection import ExplainCollection
 import sys
 from bson.son import SON
 
-class CommandLogger(monitoring.CommandListener):
-    def __init__(self):
-        self.payloads = []
+old_update_one = Collection.update_one
+old_replace_one = Collection.replace_one
+old_update_many = Collection.update_many
+old_delete_one = Collection.delete_one
+old_delete_many = Collection.delete_many
+old_aggregate = Collection.aggregate
+old_watch = Collection.watch
+old_find = Collection.find
+old_find_one = Collection.find_one
+old_find_one_and_delete = Collection.find_one_and_delete
+old_find_one_and_replace = Collection.find_one_and_replace
+old_find_one_and_update = Collection.find_one_and_update
+old_count_documents = Collection.count_documents
+old_estimated_document_count = Collection.estimated_document_count
+old_distinct = Collection.distinct
 
-    def started(self, event):
-        print("started")
-        self.payloads.append(event.command)
+old_functions = Collection.update_one
+old_replace_one = Collection.replace_one,
+old_update_many = Collection.update_many,
+old_delete_one = Collection.delete_one,
+old_delete_many = Collection.delete_many,
+old_aggregate = Collection.aggregate,
+old_watch = Collection.watch,
+old_find = Collection.find,
+old_find_one = Collection.find_one,
+old_find_one_and_delete = Collection.find_one_and_delete,
+old_find_one_and_replace = Collection.find_one_and_replace,
+old_find_one_and_update = Collection.find_one_and_update,
+old_count_documents = Collection.count_documents,
+old_estimated_document_count = Collection.estimated_document_count,
+old_distinct = Collection.distinct
 
-    def succeeded(self, event):
-        pass
+def update_one(self: Collection, filter, update, upsert=False,
```

Comment:
What do you think about doing this instead?
```
def update_one(self, *args, **kwargs):
    ExplainCollection(self).update_one(*args, **kwargs)
    return old_update_one(self, *args, **kwargs)
```

Reply:
Done.